slick/src/main/scala/slick/jdbc/H2Profile.scala

```diff
@@ -123,6 +127,32 @@ trait H2Profile extends JdbcProfile {
       override def valueToSQLLiteral(value: UUID) = "'" + value + "'"
       override def hasLiteralForm = true
     }
+    override val instantType : InstantJdbcType = new InstantJdbcType {
+      val formatter = new DateTimeFormatterBuilder()
+                      .append(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"))
+                      .optionalStart()
+                      .appendFraction(ChronoField.NANO_OF_SECOND, 0, 9, true)
+                      .optionalEnd()
+                      .appendOffset("+HH", "")
+                      .toFormatter()
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "TIMESTAMP(9) WITH TIME ZONE"
```

Comment:
As instants should always be in the UTC zone offset, storing the timezone is not necessary